pkg/platform/kube/functionres/lazy.go

```diff
@@ -2184,23 +2191,78 @@ func (lc *lazyClient) resolveFailFast(pods []v1.Pod,
 				}
 			}
 		}
+	}
 
+	// check each pod's conditions to determine if there is at least one unschedulable pod
+	errGroup, _ := errgroup.WithContext(context.TODO(), lc.logger)
+	lock := sync.Mutex{}
+	scaleUpOccurred := false
+	for _, pod := range pods {
+		pod := pod
 		for _, condition := range pod.Status.Conditions {
 
 			// check if the pod is in pending state, and the reason is that it is unschedulable
 			// (meaning no k8s node can currently run it, because of insufficient resources etc..)
 			if pod.Status.Phase == v1.PodPending && condition.Reason == "Unschedulable" {
 
-				// TODO: if pods has an event that cluster is going to scale up - don't return error
-
-				return errors.Errorf("NuclioFunction pod (%s) is unschedulable", pod.Name)
+				errGroup.Go("WaitAndCheckAutoScaleEvents", func() error {
+					time.Sleep(15 * time.Second)
+
+					// check if the pod is unschedulable due to scaling up
+					triggeredScaleUp, err := lc.isPodAutoScaledUp(pod)
+					if err != nil {
+						return errors.Wrapf(err,
+							"Failed to resolve pod (%s) triggered a node scale up",
+							pod.Name)
+					}
+					if triggeredScaleUp {
+						lc.logger.InfoWith("Nuclio function pod has triggered a scale up", "podName", pod.Name)
+						lock.Lock()
+						defer lock.Unlock()
+
+						scaleUpOccurred = true
+					} else {
+						return errors.Errorf("NuclioFunction pod (%s) is unschedulable", pod.Name)
+					}
+					return nil
+				})
 			}
 		}
 	}
+	if err := errGroup.Wait(); err != nil {
+		return errors.Wrap(err, "Failed to verify at least one pod schedulability")
+	}
+	if scaleUpOccurred {
 
+		// TODO: Increase wait timeout?
+		lc.logger.Debug("Scale up occurred")
```

Comment:
well, not really. what did happen is that the pod *triggered* a scale up. until pod is scheduled, you dont know if the scale up succeeded or not. and matter of fact it doesnt matter. what matter to us is whether the pod is scheduled or not.
in this case, when scale up is determined, you can _respect_ the readinessTimeout without fast-failing, that's it.

Reply:
Other than logging that, do we want to do anything else in this case?

Reply:
not really. the readiness timeout should be respected. the fail-fast should not fail when scale up occurs.